server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -301,24 +302,36 @@ public void removeResources(PurgeEvent event) {
 
   private void cleanupStorageSelectionCache(PurgeEvent event) {
     Function<String, Boolean> deleteConditionFunc = null;
+    String prefixKey = null;
     if (event instanceof AppPurgeEvent) {
+      prefixKey = UnionKey.buildKey(event.getAppId());
       deleteConditionFunc =
           partitionUnionKey -> UnionKey.startsWith(partitionUnionKey, event.getAppId());
     } else if (event instanceof ShufflePurgeEvent) {
+      assert event.getShuffleIds().size() == 1; // only one shuffleId
```

Comment:
Could you log the size  and ignore the other ids, rather than `assert` ? This is harmful for a running service